### PR TITLE
[ONNX] Update `unconvertible_ops`

### PR DESCRIPTION
**Pull request by justinchuby**
Update `unconvertible_ops` to create a list of unconvertible ops using the updated registry.

- Use fewer passes in the jit process instead to avoid errors during conversion in the ONNX fallback mode
- Actually check the registry to find implemented ops
- Fix type hints for `_create_jit_graph` and `_jit_pass_onnx_remove_inplace_ops_for_onnx`

### CODE REVIEW

**Comment by justinchuby** [comment]
Looks like 

https://github.com/pytorch/pytorch/blob/41be45f0f4c6db2755be907db4f4a1665fe312e0/test/onnx/test_utility_funs.py#L72-L87

needs to be updated

**Comment by justinchuby** [comment]
@yrajas

File: torch/onnx/utils.py [hunk 1]
```diff
@@ -1245,42 +1240,68 @@ def export_to_pretty_string(
...
+            # This way we can always generate a list just by looking at the names
+            # of the ops in the graph.
+            args = _decide_input_format(model, args)
+            model = _pre_trace_quant_model(model, args)
+            graph, _, _, module = _create_jit_graph(model, args)
+            _C._jit_pass_inline(graph)
+            _C._jit_pass_onnx_remove_inplace_ops_for_onnx(graph, module)
+            _C._jit_pass_erase_number_types(graph)
+            _C._jit_pass_dce_allow_deleting_nodes_with_side_effects(graph)
+    except Exception as e:
```
Comment (AllenTiTaiWang) [inline torch/onnx/utils.py:1284]: Did you bump into certain error that enters here oftenly?

File: torch/onnx/utils.py [hunk 2]
```diff
@@ -1245,42 +1240,68 @@ def export_to_pretty_string(
...
+            # Create a mostly clean JIT graph that contains the plain aten and
+            # other ops we can check with the symbolic registry.
+            # NOTE: We don't want to actually convert any ops to ONNX or run any
+            # symbolic functions because there is a higher chance that a pass
+            # fails or an unconvertible op messes up the graph during ONNX conversion.
+            # This way we can always generate a list just by looking at the names
+            # of the ops in the graph.
+            args = _decide_input_format(model, args)
+            model = _pre_trace_quant_model(model, args)
+            graph, _, _, module = _create_jit_graph(model, args)
```
Comment (AllenTiTaiWang) [inline torch/onnx/utils.py:1279]: How do you make sure that slipping passes won't make error in the graph? Whether it passes CI?

File: torch/onnx/utils.py [hunk 3]
```diff
@@ -1245,42 +1240,68 @@ def export_to_pretty_string(
...
-        if node.kind().split(":")[0] not in supported_namespaces:
-            unsupported_ops.append(node.kind())
+        domain_op = node.kind()
+        if domain_op.startswith("onnx::") or domain_op.startswith("prim::"):
+            # We consider onnx and prim ops as supported ops, even though some "prim"
+            # ops are not implemented as symbolic functions, because they may be
+            # eliminated in the conversion passes. Users may still see errors caused
+            # by prim ops even though they don't show up in the list.
+            continue
+        if not registration.registry.is_registered_op(domain_op, opset_version):
```
Comment (AllenTiTaiWang) [inline torch/onnx/utils.py:1299]: I am curious if we can simply use torch graph only with registry to check unconvertible ops?

Comment (justinchuby) [reply torch/onnx/utils.py:1299]: Yeah. This is the point where we have access to the ops where the graph is flattened.

**Review by BowenBao — APPROVED** [review]
LGTM. This slightly changes the behavior of previous version. This is a useful tool until we are able to emit such information via `export` call.

**Pull request closed without merging**